mk/dist.mk

```diff
@@ -87,10 +88,11 @@ $(PKG_TAR): $(PKG_FILES)
          --exclude=*/llvm/test/*/*/*.ll \
          --exclude=*/llvm/test/*/*/*.td \
          --exclude=*/llvm/test/*/*/*.s \
-         -c $(UNROOTED_PKG_FILES) | tar -x -f - -C tmp/dist/$(PKG_NAME)
+         -c $(UNROOTED_PKG_FILES) | tar -x -f - -C tmp/dist/$(SRC_PKG_NAME)-image/lib/rust-lib/rust-src
+
+$(PKG_TAR): tmp/dist/$$(SRC_PKG_NAME)-image
 	@$(call E, making $@)
-	$(Q)tar -czf $(PKG_TAR) -C tmp/dist $(PKG_NAME)
-	$(Q)rm -Rf tmp/dist/$(PKG_NAME)
+	$(Q)tar -czf $(PKG_TAR) -C tmp/dist/$(SRC_PKG_NAME)-image/lib/rust-lib rust-src --transform 's,^rust-src,$(PKG_NAME),S'
```

Comment:
Both source packages are built from the same image, and the directory name is `rust-src`, but `tar` uses the directory name as the root directory in the archive, whereas we used to name the root directory `$(PKG_NAME)`, so the `--transform` option rewrites all the paths so that the resulting .tar.gz matches exactly the format we used before.